src/devices/UFireIse/UFireIse.cs

```diff
@@ -0,0 +1,415 @@
+        {
+            WriteRegister(Register.ISE_TEMP_REGISTER, Convert.ToSingle(temp.DegreesCelsius));
+            _temperature = temp;
+        }
+
+        /// <summary>
+        /// Calibrates the probe using a single solution. Put the probe in a solution where the pH (Power of Hydrogen) is know.
+        /// </summary>
+        /// <param name="solution">The known pH value in mV. Range: -1024 mV to 1024 mV</param>
+        public void CalibrateFromSingleValue(ElectricPotential solution)
+        {
+            if (solution.Millivolts >= -1024 && solution.Millivolts <= 1024)
+            {
+                WriteRegister(Register.ISE_SOLUTION_REGISTER, Convert.ToSingle(solution.Millivolts));
+                SendCommand((byte)Command.ISE_CALIBRATE_SINGLE);
+
+                DelayHelper.DelayMilliseconds(IseMvMeasureTime, allowThreadYield: true);
+            }
+
+            throw new ArgumentOutOfRangeException("The value range is -1024 mV to 1024 mV");
+        }
+
+        /// <summary>
+        /// The lower value when calibrating from two solutions. Put the probe in a solution with the lowest pH (Power of Hydrogen) value.
+        /// </summary>
+        /// <param name="solution">The known pH value in mV. Range: -1024 mV to 1024 mV</param>
+        public void CalibrateFromTwoValuesLowValue(ElectricPotential solution)
+        {
+            if (solution.Millivolts >= -1024 && solution.Millivolts <= 1024)
+            {
+                WriteRegister(Register.ISE_SOLUTION_REGISTER, Convert.ToSingle(solution.Millivolts));
+                SendCommand((byte)Command.ISE_CALIBRATE_LOW);
+
+                DelayHelper.DelayMilliseconds(IseMvMeasureTime, allowThreadYield: true);
+            }
+
+            throw new ArgumentOutOfRangeException("The value range is -1024 mV to 1024 mV");
+        }
+
+        /// <summary>
+        ///  The highest value when calibrating from two solutions. Put the probe in a solution with the highest pH (Power of Hydrogen) value.
+        /// </summary>
+        /// <param name="solution">The known pH value in mV. Range: -1024 mV to 1024 mV</param>
+        public void CalibrateFromTwoValuesHighValue(ElectricPotential solution)
+        {
+            if (solution.Millivolts >= -1024 && solution.Millivolts <= 1024)
+            {
+                WriteRegister(Register.ISE_SOLUTION_REGISTER, Convert.ToSingle(solution.Millivolts));
+                SendCommand((byte)Command.ISE_CALIBRATE_HIGH);
+
+                DelayHelper.DelayMilliseconds(IseMvMeasureTime, allowThreadYield: true);
+            }
+
+            throw new ArgumentOutOfRangeException("The value range is -1024 mV to 1024 mV");
+        }
+
+        /// <summary>
+        /// Returns the firmware version of the device.
+        /// </summary>
+        /// <returns>Firmware version</returns>
+        public Version GetVersion()
+        {
+            return new Version((int)ReadFloat((byte)Register.ISE_VERSION_REGISTER), 0);
```

Comment:
I'm surprised the version is float here... should this be integer?

Reply:
I have changed it back to a string, because the manufacturer do not provide any information about the format of the version number. This insures it supports any format it sends back.

Reply:
They seem to leave it as byte: https://github.com/u-fire/Isolated_ISE/blob/b2d58b40a7873cda98796f714272695bec365cea/python/RaspberryPi/uFire_ISE.py#L97

You can either assume it's major 0 and minor = whatever_you_read or swap minor and major, up to you but be consistent

Reply:
@justind000 do you perhaps know if the version in ISE_VERSION_REGISTER is major or minor? How should we interpret it? (or doesn't really matter?)

Reply:
if we look at https://github.com/u-fire/ISE_Probe/blob/master/python/RaspberryPi/shell.py#L91 the GetVersion is major  and GetFirmwareVersion is minor. That is why I first made GetVersion return major  and GetFirmwareVersion  return minor.

Reply:
@krwq I have pushed a new version where GetVersion return major and GetFirmwareVersion return  minor. Is this ok, or do I need to change it, sow both return major or minor?

Reply:
be consistent, either both minor of both major

Reply:
not sure in that case... perhaps we should join both versions together into a single `GetVersion`? Honestly I'd be fine if you returned byte from both methods at this point 😄

Reply:
it now return byte